### DISCUSSION TITLE
Support bazel for ros-foxglove-bridge with ROS2

### DISCUSSION THREAD

**Discussion by kgreenek** [foxglove/community#684] [General]
Our project is using ros2 with bazel.

It would be convenient for us to be able to deploy ros-foxglove-bridge along with the rest of our stack using bazel instead of having to dpeloy it in a separate docker container that has ros-foxlove-bridge system-installed with Ubuntu.

There are excellent bazel rules for ROS2 here: https://github.com/mvukov/rules_ros2

I would be happy to contribute this upstream if it is something others would use too.

Cheers!

**Comment by defunctzombie** [comment]
Is there a repository of bazel rules for ROS packages? Our preference would be to have the rules live there since bazel is not a first-class target for ROS packages and we ourselves are not actively using bazel or looking to take on the maintenance of ensuring the bazel files work. If you setup such a repository we would be happy to reference it from the README as an additional reference.

**Comment by kgreenek** [comment]
Not that I am aware of, but I can just host a separate repo with just the
bazel rules for ros-foxglove-bridge. Thank you.

**Comment by mvukov** [comment]
Implemented in https://github.com/mvukov/rules_ros2/pull/113.

**Reply by jtbandes** [reply]
Very cool, thanks for sharing!
